macroid-core/src/main/scala/macroid/MultiEventTweaks.scala

```diff
@@ -0,0 +1,146 @@
+package macroid
+
+import scala.language.dynamics
+import scala.language.experimental.macros
+import android.view.View
+import scala.reflect.macros.{Context ⇒ MacroContext}
+import scala.language.postfixOps
+
+private[macroid] trait MultiEventTweaks {
+
+  import MultiEventTweakMacros._
+
+  type UnitHandler = String
+
+  type FuncHandler = String
+
+  sealed trait HandlerType
+
+  case object HUnit extends HandlerType
+
+  case object HUnitFunc extends HandlerType
+
+  object UnitOn extends Dynamic {
+    def applyDynamic[W <: View](event: String)(handler: Any): Tweak[W] = macro onUnitImpl[W]
+  }
+
+  object UnitFuncOn extends Dynamic {
+    def applyDynamic[W <: View](event: String)(handler: Any): Tweak[W] = macro onUnitFuncImpl[W]
+  }
+
+  object MultiOn extends Dynamic {
+    def applyDynamic[W <: View](event: String)(handlers: (String => Any)*): Tweak[W] = macro onMultipleFuncImpl[W]
+  }
+}
+
+object MultiEventTweaks extends MultiEventTweaks
+
+private object MultiEventTweakMacros {
+
+  import MultiEventTweaks._
+
+  def onUnitImpl[W <: View : c.WeakTypeTag](c: MacroContext)(event: c.Expr[String])(handler: c.Expr[Any]) = {
+    import c.universe._
+
+    val Expr(Literal(Constant(e: String))) = event
+
+    onMultipleFuncImpl(c)(event)(c.Expr[UnitHandler => Any](q"(${newTermName(s"on${e.capitalize}")}: UnitHandler) => $handler"))
+  }
+
+  def onUnitFuncImpl[W <: View : c.WeakTypeTag](c: MacroContext)(event: c.Expr[String])(handler: c.Expr[Any]) = {
+    import c.universe._
+
+    val Expr(Literal(Constant(e: String))) = event
+
+    onMultipleFuncImpl(c)(event)(c.Expr[FuncHandler => Any](q"(${newTermName(s"on${e.capitalize}")}: FuncHandler) => $handler"))
+  }
+
+  def onMultipleFuncImpl[W <: View : c.WeakTypeTag](c: MacroContext)(event: c.Expr[String])(handlers: (c.Expr[String => Any])*) = {
+    import c.universe._
+
+    val tpe = weakTypeOf[W]
+    val (_, setter) = findSetter(c)(event, tpe)
+
+    val (listener, handlerOnMethodTupleSequence) = onBaseMultipleHandlers(c)(setter, handlers)
+
+    val overrides = handlerOnMethodTupleSequence map { case (handler, on, handlerType) =>
+      getOverride(c)(on, c.Expr(c.resetLocalAttrs(handler.tree.children.last)), handlerType)
+    }
+
+    c.Expr[Tweak[View]](q"_root_.macroid.Tweak[$tpe] { x ⇒ x.$setter(new $listener { ..$overrides })}")
+  }
+
+  def findSetter(c: MacroContext)(event: c.Expr[String], tp: c.Type) = {
```

Comment:
This method is an almost verbatim copy of the code from `EventTweakMacros.onBase`. I think it would be better to reuse the same method in both cases — could be either this one or that one.